src/bots/userLpSettler.ts

```diff
@@ -0,0 +1,383 @@
+				this.watchdogTimerLastPatTime = Date.now();
+			});
+		}
+	}
+
+	async trySendTxForChunk(ixs: TransactionInstruction[]): Promise<void> {
+		const success = await this.sendTxForChunk(ixs);
+		if (!success) {
+			const slice = ixs.length / 2;
+			if (slice < 1) {
+				await webhookMessage(
+					`[${this.name}]: :x: Failed to settle LPs, reduced until 0 ixs...`
+				);
+				return;
+			}
+			await sleepMs(SLEEP_MS);
+			await this.sendTxForChunk(ixs.slice(0, slice));
+			await sleepMs(SLEEP_MS);
+			await this.sendTxForChunk(ixs.slice(slice));
+		}
+		await sleepMs(SLEEP_MS);
+	}
+
+	async sendTxForChunk(ixs: TransactionInstruction[]): Promise<boolean> {
+		if (ixs.length == 0) {
+			return true;
+		}
+
+		let success = false;
+		logger.info(
+			`Using avgPriorityFee: ${
+				this.priorityFeeSubscriber!.lastAvgStrategyResult
+			} (clamp to ${MAX_COMPUTE_UNIT_PRICE_MICRO_LAMPORTS})`
+		);
+		try {
+			const simResult = await simulateAndGetTxWithCUs(
+				ixs,
+				this.driftClient.connection,
+				this.driftClient.txSender,
+				[this.lookupTableAccount!],
+				[],
+				undefined,
+				1.15,
+				true,
+				true
+			);
+			logger.info(
+				`Settle LP estimated ${simResult.cuEstimate} CUs for ${ixs.length} settle LPs.`
+			);
+
+			if (simResult.simError !== null) {
+				logger.error(
+					`Sim error: ${JSON.stringify(simResult.simError)}\n${
+						simResult.simTxLogs ? simResult.simTxLogs.join('\n') : ''
+					}`
+				);
+				success = false;
+			} else {
+				const txSig = await this.driftClient.txSender.sendVersionedTransaction(
+					simResult.tx,
+					[],
+					this.driftClient.opts
+				);
+				logger.info(
+					`Settled LPs for ${ixs.length} users in tx: https://solana.fm/tx/${txSig.txSig}`
+				);
+				success = true;
+			}
+		} catch (err) {
+			console.error(err);
+			// const userKeys = users
+			// 	.map(({ settleeUserAccountPublicKey }) =>
+			// 		settleeUserAccountPublicKey.toBase58()
+			// 	)
+			// 	.join(', ');
+			// logger.error(`Failed to settle pnl for users: ${userKeys}`);
+			// logger.error(err);
+
+			// if (err instanceof Error) {
+			// 	const errorCode = getErrorCode(err) ?? 0;
+			// 	if (!errorCodesToSuppress.includes(errorCode) && users.length === 1) {
+			// 		if (err instanceof SendTransactionError) {
+			// 			await webhookMessage(
+			// 				`[${this.name
+			// 				}]: :x: Error code: ${errorCode} while settling pnls for ${marketIndex}:\n${err.logs ? (err.logs as Array<string>).join('\n') : ''
+			// 				}\n${err.stack ? err.stack : err.message}`
+			// 			);
+			// 		}
+			// 	}
+			// }
+		}
+		return success;
+	}
+
+	private async fetchLpUsers(): Promise<
+		Array<{
+			publicKey: PublicKey;
+			userAccount: UserAccount;
+		}>
+	> {
+		const rpcRequestArgs = [
+			this.driftClient.program.programId.toBase58(),
+			{
+				commitment: 'confirmed',
+				filters: [
+					getUserFilter(),
+					getNonIdleUserFilter(),
+					// getUserThatHasBeenLP()], // doesn't work
```

Comment:
should we just use user map?

Reply:
yeah, no reason not to, i just did this custom gPA to see if that filter works (doesnt seem to)